Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp

```diff
@@ -1180,15 +1180,41 @@ WebIDL::ExceptionOr<void> HTMLInputElement::set_size(unsigned value)
     return set_attribute(HTML::AttributeNames::size, MUST(String::number(value)));
 }
 
+// https://html.spec.whatwg.org/multipage/input.html#concept-input-value-string-number
+Optional<double> HTMLInputElement::convert_string_to_number(StringView input) const
+{
+    // https://html.spec.whatwg.org/multipage/input.html#number-state-(type=number):concept-input-value-string-number
+    if (type_state() == TypeAttributeState::Number) {
+        auto maybe_number = parse_floating_point_number(input);
+        if (maybe_number.has_value())
+            return maybe_number.value();
+        return {};
+    }
+
+    // FIXME: Implement more input types
+    VERIFY_NOT_REACHED();
```

Comment:
If we're going to crash, use `TODO` instead of `VERIFY_NOT_REACHED`. But it's generally less annoying to just add a `dbgln` with the type state to be implemented (and return an empty `Optional` in this case).